x/config/loader.go

```diff
@@ -0,0 +1,282 @@
+	return s.onewayOutboundConfigType != nil
+}
+
+func (s *transportSpec) transportBuilder(attrs attributeMap) (TransportBuilder, error) {
+	result := reflect.New(s.transportConfigType).Interface()
+	if err := attrs.Decode(result); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration for transport %q: %v", s.name, err)
+	}
+	return result.(TransportBuilder), nil
+}
+
+func (s *transportSpec) inboundBuilder(attrs attributeMap) (InboundBuilder, error) {
+	if s.inboundConfigType == nil {
+		return nil, fmt.Errorf("transport %q does not define an inbound", s.name)
+	}
+
+	result := reflect.New(s.inboundConfigType).Interface()
+	if err := attrs.Decode(result); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration for inbound %q: %v", s.name, err)
+	}
+	return result.(InboundBuilder), nil
+}
+
+func (s *transportSpec) unaryOutboundBuilder(preset string, attrs attributeMap) (UnaryOutboundBuilder, error) {
+	typ := s.unaryOutboundConfigType
+	if typ == nil {
+		return nil, fmt.Errorf("transport %q does not support unary outbounds", s.name)
+	}
+
+	if preset != "" {
+		var ok bool
+		typ, ok = s.unaryOutboundPresets[preset]
+		if !ok {
+			return nil, fmt.Errorf("unknown preset %q for unary outbound %q", preset, s.name)
+		}
+	}
+
+	result := reflect.New(typ).Interface()
+	if err := attrs.Decode(result); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration for unary outbound %q: %v", s.name, err)
+	}
+	return result.(UnaryOutboundBuilder), nil
+}
+
+func (s *transportSpec) onewayOutboundBuilder(preset string, attrs attributeMap) (OnewayOutboundBuilder, error) {
+	typ := s.onewayOutboundConfigType
+	if typ == nil {
+		return nil, fmt.Errorf("transport %q does not support oneway outbounds", s.name)
+	}
+
+	if preset != "" {
+		var ok bool
+		typ, ok = s.onewayOutboundPresets[preset]
+		if !ok {
+			return nil, fmt.Errorf("unknown preset %q for oneway outbound %q", preset, s.name)
+		}
+	}
+
+	result := reflect.New(typ).Interface()
+	if err := attrs.Decode(result); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration for oneway outbound %q: %v", s.name, err)
+	}
+	return result.(OnewayOutboundBuilder), nil
+}
+
+// Loader TODO
+type Loader struct {
+	knownTransports map[string]transportSpec
+}
+
+// NewLoader TODO
+func NewLoader() *Loader {
+	return &Loader{knownTransports: make(map[string]transportSpec)}
+}
+
+func (l Loader) spec(name string) (transportSpec, error) {
+	spec, ok := l.knownTransports[name]
+	if !ok {
+		return transportSpec{}, fmt.Errorf("unknown transport %q", name)
+	}
+	return spec, nil
+}
+
+// LoadYAML loads a YARPC configuration from YAML.
+func (l *Loader) LoadYAML(b []byte) (*YARPC, error) {
+	var data map[string]interface{}
+	if err := yaml.Unmarshal(b, &data); err != nil {
+		return nil, err
+	}
+	return l.Load(data)
+}
+
+// Load a YARPC configuration from the given data map.
+func (l *Loader) Load(data map[string]interface{}) (*YARPC, error) {
```

Comment:
Yeah, this is the function you will call after you have created the final merged config with your own YAML parsing logic.